packages/vike-node/src/runtime/vike-handler.ts

```diff
@@ -33,24 +33,88 @@ async function renderPage<PlatformRequest>({
   return pageContext.httpResponse
 }
 
-async function renderPageWeb<PlatformRequest>({
-  url,
-  headers,
-  platformRequest,
-  options
-}: {
-  url: string
-  headers: [string, string][]
-  platformRequest: PlatformRequest
-  options: VikeOptions<PlatformRequest>
-}) {
-  const httpResponse = await renderPage({
-    url,
-    headers,
-    platformRequest,
-    options
+export const compressMiddleware = ((options?) => async (request, _context, runtime: any) => {
+  const compressionType = options?.compress ?? !isVercel()
+  const compressMiddlewareInternal = compressMiddlewareFactory()(request)
+
+  return async (response) => {
+    if (!globalStore.isDev) {
+      const isAsset = new URL(request.url).pathname.startsWith('/assets/')
+      const shouldCompressResponse = compressionType === true || (compressionType === 'static' && isAsset)
+      if (shouldCompressResponse) {
+        return compressMiddlewareInternal(response)
+      }
+    }
+    return response
+  }
+}) satisfies Get<[options: VikeOptions], UniversalMiddleware>
+
+export const renderPageHandler = ((options?) => async (request, context, runtime: any) => {
+  const nodeReq: IncomingMessage | undefined = runtime.req
+  let staticConfig: false | { root: string; cache: boolean } = false
+  let staticMiddleware: ConnectMiddleware | undefined
+
+  if (nodeReq) {
+    globalStore.setupHMRProxy(nodeReq)
+    const { resolveStaticConfig } = await import('./utils/resolve-static-config.js')
+    staticConfig = resolveStaticConfig(options?.static)
+  }
+
+  if (globalStore.isDev) {
+    const handled = await web(request)
+
+    if (handled) return handled
+  } else if (nodeReq) {
+    if (staticConfig) {
+      const handled = await connectToWeb(serveStaticFiles)(request)
+      if (handled) return handled
+    }
+  }
+
+  async function serveStaticFiles(req: IncomingMessage, res: ServerResponse): Promise<boolean> {
+    if (!staticMiddleware) {
+      const { default: sirv } = await import('sirv')
+      staticMiddleware = sirv((staticConfig as { root: string; cache: boolean }).root, { etag: true })
+    }
+
+    return new Promise<boolean>((resolve) => {
+      res.once('close', () => resolve(true))
+      staticMiddleware!(req, res, () => resolve(false))
+    })
+  }
+
+  const pageContextInit = { ...context, ...runtime, urlOriginal: request.url, headersOriginal: request.headers }
+  const response = await renderPage({
+    url: request.url,
+    headers: parseHeaders(request.headers),
+    options: {
+      ...options,
+      pageContext: {
+        ...pageContextInit,
+        ...options?.pageContext
+      }
+    }
+  })
+
+  const { readable, writable } = new TransformStream()
+  response.pipe(writable)
```

Comment:
How about we use `pageContext.httpResponse.getReadableWebStream()`? (I updated the recommendation at https://vike.dev/streaming#basics to be more accurate.)